scipy/optimize/_linprog.py

```diff
@@ -581,7 +581,7 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
         warning_message = "x0 is used only when method is 'revised simplex'. "
         warn(warning_message, OptimizeWarning)
 
-    if integrality and not meth == "highs":
```

Comment:
This isn't equivalent, right? 

```python
>>> a = [0, 0]
>>> if a:
...   print("here")
...
here
>>> if np.any(a):
...   print("here 2")
...
>>>
```

Reply:
No, but it's appropriate. It's just a little less noisy. Previously, it would warn if the user passed [0, 0], but explicitly stating that there are no inequality constraints like this doesn't affect the problem, so there's no real need to warn.